rcldotnet/RCLdotnet.cs

```diff
@@ -198,9 +210,17 @@ public class RCLdotnet {
 
     public static Node CreateNode (string nodeName, string nodeNamespace) {
       IntPtr nodeHandle = IntPtr.Zero;
-      int ret = RCLdotnetDelegates.native_rcl_create_node_handle (ref nodeHandle, nodeName, nodeNamespace);
-      Node node = new Node (nodeHandle);
-      return node;
+      RCLRet ret = (RCLRet)RCLdotnetDelegates.native_rcl_create_node_handle (ref nodeHandle, nodeName, nodeNamespace);
+
+      if (ret == RCLRet.Ok)
+      {
+        Node node = new Node (nodeHandle);
+        return node;
+      } else if (ret == RCLRet.NodeInvalidName) {
+        throw new InvalidNodeNameException(nodeName, RMWGetErrorString());
+      } else {
+        throw new Exception();
```

Comment:
Throwing a plain `Exception` here doesn't give enough information about what went wrong. Could you add an exception for each case that could fail in `rcl_create_node_handle`?

Reply:
I added an exception for each return value of `rmw_validate_node_name` and `rcl_node_init`